packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -172,7 +172,9 @@ export class WebChannelConnection implements Connection {
         }
       });
 
-      const requestString = JSON.stringify(request);
+      const jsonObj = { ...request } as any;
+      delete jsonObj.database; // The database field is already encoded in URL.
```

Comment:
Your comment makes it sound like we are only doing this as an optimization, but as I understand it, this is _required_ to make the SDK work with the emulator and so I would make that clearer in the comment so somebody modifying this code in the future knows the consequences of changing this.

Reply:
Good point! I've added more comments.